source/tests/system/nirfmxinstr_driver_api_tests.cpp

```diff
@@ -165,6 +165,28 @@ TEST_F(NiRFmxInstrDriverApiTests, SetAndGetFrequencyReferenceSource_ReturnsFrequ
   EXPECT_EQ(response.attr_val(), "OnboardClock");
 }
 
+TEST_F(NiRFmxInstrDriverApiTests, GetModelName_ReturnsModelName)
+{
+  const auto session = init_session(stub(), PXI_5663E);
+  const auto response = client::get_attribute_string(stub(), session, "", NiRFmxInstrAttribute::NIRFMXINSTR_ATTRIBUTE_INSTRUMENT_MODEL);
+
+  EXPECT_SUCCESS(session, response);
+  EXPECT_EQ(response.attr_val(), "NI PXIe-5663E");
+}
+
+TEST_F(NiRFmxInstrDriverApiTests, TimestampFromValuesRoundTrip_SucceedsWithOriginalValues)
+{
+  constexpr auto SECONDS_SINCE_1970 = 10000;
+  constexpr auto FRACTIONAL_SECONDS = .75;
+  const auto timestamp_response = client::timestamp_from_values(stub(), SECONDS_SINCE_1970, FRACTIONAL_SECONDS);
+  const auto values_response = client::values_from_timestamp(stub(), timestamp_response.timestamp());
```

Comment:
Question: I am assuming this is a simulation hardware where internal time doesn't change?

Reply:
That's a good question. I might be misunderstanding (chances are good since time is involved).

Neither of these APIs are meant to give hardware/current/os time as far as I know. In the C API, they're convenience methods for converting input values to/from `CVIAbsoluteTime`, which is a standard NI time representation. We convert `CVIAbsoluteTime` to [protobuf Timestamp](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Timestamp), so I'm trying to validate that that (double) conversion is working.

I'm honestly not sure why these utility function are in RFmxInstr, because I don't see APIs in the other personalities that are dealing with either time representation (maybe there's an attribute that I'm missing). But it's great for testing roundtrip conversion because we don't need to rely on any state. I wish we'd started with this API when adding it in the first place (Hi @gregstoll )